data/shared_state_registry.go

```diff
@@ -14,7 +14,7 @@ import (
 )
 
 type SharedStateRegistry interface {
-	GetCounterInt64(key string) CounterInt64SharedVariable
+	GetCounterInt64(key string, maxAllowedDrift int64) CounterInt64SharedVariable
```

Comment:
```suggestion
	GetCounterInt64(key string, toleratedDownDrift int64) CounterInt64SharedVariable
```

To be more precise (only for downwards decreases)?